packages/material-ui/src/Portal/Portal.js

```diff
@@ -17,65 +17,60 @@ function getOwnerDocument(element) {
  * Portals provide a first-class way to render children into a DOM node
  * that exists outside the DOM hierarchy of the parent component.
  */
-class Portal extends React.Component {
-  componentDidMount() {
-    this.setMountNode(this.props.container);
+function Portal(props) {
+  const {children, disablePortal, onRendered, container} = props;
+  const [, forceUpdate] = React.useState(0);
+  const mountNode = React.useRef();
+  const renderedTimer = React.useRef();
+
+  const setMountNode = React.useCallback(() => {
+    if (disablePortal) {
+      mountNode.current = ReactDOM.findDOMNode(this).parentElement;
```

Comment:
We would need to add a ref to the child and require that it forwards the ref to a DOM component. Function components don't have an instance which is why `findDOMNode(this)` won't work.

Reply:
Hi @eps1lon 
Just wanted to confirm if I understand it correctly, are you trying to say something like this:
```diff
....
....
const setMountNode = React.useCallback(
    () => {
      if (disablePortal) {
+        mountNode.current = ReactDOM.findDOMNode(props.innerRef).parentElement;
-         mountNode.current = ReactDOM.findDOMNode(this).parentElement;
        return;
      }
+      mountNode.current = getContainer(container, getOwnerDocument(props.innerRef).body);
-      mountNode.current = getContainer(container, getOwnerDocument(this).body);
    },
    [container, disablePortal],
  );
....
....
+const child = React.cloneElement(children, {
+    ref: props.innerRef,
+    ...children.props,
+  });

 return mountNode.current ? ReactDOM.createPortal(child, mountNode.current) : null
.....
.....
+export default React.forwardRef((props, ref) => <Portal innerRef={ref} {...props} />);

Reply:
Kind of. You have to declare the ref in the component though. You can take a look at #15344 how you should attach a ref to children.

Reply:
Hi @eps1lon 
I am a bit of a beginner when it comes to React hooks. I am not able to understand correctly the failing test. Can you please help me out here.

Thanks 😃